### PR TITLE
Make MultiDMTestCases be data driven by using JUnit Parameterized Tests.

### PR DESCRIPTION
**Pull request by quinton-hoole**
Rather than write an explicit test for every possible combination of DM's, here I generate the combinations to programatically create the tests.

Details of Parameterized Tests are here:

https://dzone.com/articles/junit-parameterized-test

140 such autogenerated tests now pass.  
An additional 30 are explicitly disabled, mostly due to two broken DMs.

### CODE REVIEW

**Comment by quinton-hoole** [comment]
Made some more progress on this.  
Just need to populate the arrays with data now, and do final testing.  
20 tests are auto-generated so far, and they all pass.

File: core/src/integrationTest/java/amino/run/multidm/MultiDMTestCases.java [hunk 1]
```diff
@@ -73,7 +92,55 @@ public void setUp() throws Exception {
...
+        Arrays.sort(ignoredCombinations, comparator);
+        logger.info("Ignored combinations: " + Arrays.toString(ignoredCombinations));
+        // Add all combinations of two DMs
+        for (Class first : allDmClasses) {
+            for (Class second : allDmClasses) {
+                if (!first.equals(second)) { // Don't test DMs in combination with themselves
+                    Class[] combo = new Class[] {first, second};
+                    int foundIndex = Arrays.binarySearch(ignoredCombinations, combo, comparator);
+                    logger.info("Found Index: " + foundIndex + " for " + first.getName() + ", " + second.getName());
+                    if (Arrays.binarySearch(ignoredCombinations, combo, comparator) < 0) {
```
Comment (VenuReddy2103) [inline core/src/integrationTest/java/amino/run/multidm/MultiDMTestCases.java]: nit: We can use above foundIndex here

**Comment by quinton-hoole** [comment]
Complete.  140 MultiDM tests pass consistently now. 
The ones that don't are explicitly listed to aid debugging.
@VenuReddy2103 or @maheshrajus or @AmitRoushan can I ask you to do one final review please?

File: core/src/integrationTest/java/amino/run/multidm/MultiDMTestCases.java (hunk 1 above)
Comment (quinton-hoole) [reply core/src/integrationTest/java/amino/run/multidm/MultiDMTestCases.java]: Agreed, fixed.

**Comment by VenuReddy2103** [comment]
LGTM

**Comment by AmitRoushan** [comment]
LGTM

**Pull request merged**